tests/rbd_mirror/rbd_mirror_utils.py

```diff
@@ -887,3 +896,151 @@ def mirror_daemon_status(self, daemon):
+    Args:
+        **kw:
+
+    Examples: In default configuration, pool and image names will be taken as random values.
+        Default configuration for ecpools only :
+            config:
+                ec-pool-only: True
+        Default configuration for replicated pools only :
+            config:
+                rep-pool-only: True
+        Advanced configuration:
+            config:
+               do_not_create_image: True  # if not set then images will be created by default
+               ec-pool-k-m: 2,1
+               ec-pool-only: False
+               ec_pool_config:
+                  pool: rbd_pool_4
+                  data_pool: rbd_ec_pool_4
+                  ec_profile: rbd_ec_profile_4
+                  image: rbd_image_4
+                  size: 10G
+                  io_total: 1G
+                  mode: image (default pool)
+                  mirrormode: snapshot (default journal)
+               rep_pool_config:
+                  pool: rbd_rep_pool
+                  image: rbd_rep_image
+                  size: 10G
+                  io_total: 1G
+                  mode: image (default pool)
+                  mirrormode: snapshot (default journal)
+    Args:
+        **kw:
+
+    Returns:
+        RBD mirror objects for ecpool and replicated pool
+    """
+    rbd_obj = dict()
+
+    if not kw.get("config") or not kw.get("config").get("ec-pool-only"):
+        ec_pool_k_m = kw.get("config", {}).pop("ec-pool-k-m", None)
+
+        rep_mirror1, rep_mirror2 = [
+            RbdMirror(cluster, kw.get("config", {}))
+            for cluster in kw.get("ceph_cluster_dict").values()
+        ]
+
+        if not kw.get("config"):
+            kw["config"] = {
+                "rep_pool_config": {
+                    "pool": "rep_pool_" + rep_mirror1.random_string(),
+                    "image": "rep_image_" + rep_mirror1.random_string(),
+                    "size": "1G",
+                    "io_total": "1G",
+                    "mode": "pool",
+                }
+            }
+        elif kw.get("config").get("rep_pool_config"):
+            kw["config"]["rep_pool_config"]["pool"] = kw["config"][
+                "rep_pool_config"
+            ].get("pool", "rep_pool_" + rep_mirror1.random_string())
+            kw["config"]["rep_pool_config"]["image"] = kw["config"][
+                "rep_pool_config"
+            ].get("image", "rep_image_" + rep_mirror1.random_string())
+            kw["config"]["rep_pool_config"]["size"] = kw["config"][
+                "rep_pool_config"
+            ].get("size", "1G")
+            kw["config"]["rep_pool_config"]["io_total"] = kw["config"][
+                "rep_pool_config"
+            ].get("io_total", "1G")
+            kw["config"]["ec_pool_config"]["mode"] = kw["config"][
+                "rep_pool_config"
+            ].get("mode", "pool")
+        else:
+            kw["config"]["rep_pool_config"] = {
```

Comment:
how 952 line config and this config are different?

Reply:
Added comment as discussed